tools/rosbag/src/recorder.cpp

```diff
@@ -463,9 +464,12 @@ void Recorder::snapshotTrigger(std_msgs::Empty::ConstPtr trigger) {
     queue_condition_.notify_all();
 }
 
-void Recorder::manual_split(std_msgs::Empty::ConstPtr msg) {
-    (void)msg;
-    split_bag_ = true;
+bool Recorder::manual_split(
+        [[maybe_unused]] std_srvs::Empty::Request& req, [[maybe_unused]] std_srvs::Empty::Response& res) {
+    split_requested_ = true;
+    split();
```

Comment:
doRecord is a separate thread, seems like there will be race conditions in this implementation, try conditional variables http://www.cplusplus.com/reference/condition_variable/condition_variable/